source/MaterialXRuntime/Private/Commands/PvtReparentPrimCmd.cpp

```diff
@@ -17,13 +20,44 @@ void PvtReparentPrimCmd::execute(RtCommandResult& result)
 {
     try
     {
-        RtToken resultName = _stage->reparentPrim(_path, _parentPath);
+        // Validate that the prim exists.
+        RtPrim prim = _stage->getPrimAtPath(_path);
+        if (!prim)
+        {
+            result = RtCommandResult(false, "Given path '" + _path.asString() + " does not point to a prim in this stage");
+            return;
+        }
+
+        // Validate that the new parent exists.
+        RtPrim newParent = _stage->getPrimAtPath(_newParentPath);
+        if (!newParent)
+        {
+            result = RtCommandResult(false, "Given path '" + _newParentPath.asString() + " does not point to a prim in this stage");
+            return;
+        }
+
+        // Construct the new path making sure the prim name is according to
+        // any rename that might happen during the reparent operation.
+        PvtDataHandle newParentH = PvtObject::hnd(newParent);
+        PvtPrim* newParentPtr = newParentH->asA<PvtPrim>();
+        const RtToken newName = newParentPtr->makeUniqueChildName(prim.getName());
+        RtPath newPath = _newParentPath;
+        newPath.push(newName);
```

Comment:
Same as above, we must make sure the new name in the path is the resulting unique child name in the parent. Otherwise the message we send below will not contain an correct path.